BuildMasterAutomation/Functions/Get-BMDeployment.ps1

```diff
@@ -30,9 +50,35 @@ function Get-BMDeployment
         [Object] $Session,
 
         # The deployment to get. You can pass a deployment id or object.
-        [Parameter(Mandatory)]
         [Alias('ID')]
-        [Object] $Deployment
+        [Object] $Deployment,
+
+        # The application to get deployments for. You can pass an application id, application name, or object.
+        [Object] $Application,
+
+        # The release to get deployments for. You can pass an release id, release name, or object.
+        [Object] $Release,
+
+        # The build to get deployments for. You can pass an build id, build name, or object.
+        [Object] $Build,
+
+        # The environment to get deployments for. You can pass an environment id, environment name, or object.
+        [Object] $Environment,
+
+        # The number for the release to get deployments for.
+        [String] $ReleaseNumber,
+
+        # The number for the build to get deployments for.
+        [String] $BuildNumber,
+
+        # The name of the pipeline to get deployments for.
+        [String] $PipelineName,
+
+        # The name of the pipeline stage to get deployments for.
+        [String] $PipelineStageName,
+
+        # The status of the deployments to get. Accepted values are 'pending', 'executing', 'succeeded', 'warned', or 'failed'
+        [String] $Status

```

Comment:
Add a [`[ValidateSet()]`](https://learn.microsoft.com/en-us/powershell/scripting/developer/cmdlet/validateset-attribute-declaration?view=powershell-7.3) attribute with the list of statuses.